hooks/useScroll.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef, RefObject } from "react";
+
+interface UseScrollProps {
+  handleScrollDown: () => void;
+  handleScrollUp: () => void;
+  isUsable: boolean;
+}
+/**
+ * 특정 요소에 ref를 걸고 그 요소위에서 스크롤을 위로 올리거나 내려서 이벤트를 발생시킵니다.
+
+ * @param handleScrollUp 스크롤 위로 올릴때 발생될 함수
+ * @param handleScrollDown 스크롤 아래로 내릴때 발생될 함수
+ * @param isUsable 훅을 사용할지 안할지 결정 ( 기본 true 이므로 신경안써도 무방)
+ * @returns HTML 요소에 걸 ref
+ */
+export default function useScroll<T extends HTMLElement>({
+  handleScrollDown,
+  handleScrollUp,
+  isUsable = true,
+}: UseScrollProps): RefObject<T> {
+  const ref = useRef<T>(null);
```

Comment:
```suggestion
  const ref = useRef<HTMLElement>(null);
```
타입 고수 재영님 이렇게 바로 `HTMLElement` 넣지 않고 제너릭 T 받을 때의 장점은 어떤건가요??

Reply:
아까 이거하다가 막혔었는데 이거 
자세히좀 찾아보고 말씀드리도록 하겠습니다...!